core/test/vault-test.ts

```diff
@@ -407,3 +416,38 @@ describe('Sound values are extracted', () => {
         }
     });
 });
+Object.entries(SUPPORTED_COLLATERALS).forEach(([collateralType, collateralConfig]) => {
+    describe(`Collateral vault simulation liquidation ${collateralType}`, () => {
+        before(async () => {
+            await setupRpcUrlAndGetNetworks(LOCAL_RPC_URL);
+        });
+        it('runs the simulaton', async () => {
+            let collateralOwned: BigNumber;
+            let vaultId: number;
+            try {
+                collateralOwned = await getCollateralAmountInVat(collateralType);
+            } catch (e) {
+                if (e instanceof Error && e.message.startsWith('Cannot borrow more dai with the collateral')) {
+                    return;
+                }
+                throw e;
+            }
+            try {
+                vaultId = await createVaultForCollateral(collateralType, collateralOwned, collateralConfig.decimals);
```

Comment:
Instead of passing it as additional arg it's enough to have `collateralType` arg to determine amount of decimals using `getCollateralConfigByType`